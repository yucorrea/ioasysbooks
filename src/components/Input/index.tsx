import React from 'react';
import { GestureResponderEvent, TextInputProps } from 'react-native';
import styled from 'styled-components/native';
import { Button } from '../Button';

interface Props extends TextInputProps {
  label: string;
  enableButton?: boolean;
  onPress?: (event: GestureResponderEvent) => void;
}

export function Input({ label, enableButton, onPress, ...rest }: Props) {
  return (
    <StyledContainer>
      <StyledWrapper>
        <StyledLabel>{label}</StyledLabel>
        <StyledTextInput underlineColorAndroid="transparent" {...rest} />
      </StyledWrapper>

      {enableButton && <StyledButtonInput title="Entrar" onPress={onPress} />}
    </StyledContainer>
  );
}


export const StyledContainer = styled.View`
  flex-direction: row;
  align-items: center;
  width: 100%;
  margin-bottom: 16px;
  padding: 8px 16px 8px 13px;
  border-radius: 4px;
  background: rgba(0, 0, 0, 0.32);
`;

export const StyledWrapper = styled.View`
  flex: 1;
`;

export const StyledLabel = styled.Text`
  width: 100%;
  color: ${({ theme }) => theme.colors.text_light};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export const StyledTextInput = styled.TextInput`
  width: 100%;
  margin-top: 4px;
  color: ${({ theme }) => theme.colors.background};
  font-size: 16px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export const StyledButtonInput = styled(Button)`
  width: 85px;
`;
