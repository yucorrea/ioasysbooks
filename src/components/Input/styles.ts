import styled from 'styled-components/native';
import { Button } from '../Button';

export const Container = styled.View`
  width: 100%;
  background: rgba(0, 0, 0, 0.32);
  border-radius: 4px;
  padding: 8px 16px 8px 13px;
  margin-bottom: 16px;
  flex-direction: row;
  align-items: center;
`;

export const Wrapper = styled.View`
  flex: 1;
`;

export const Label = styled.Text`
  width: 100%;
  color: ${({ theme }) => theme.colors.text_light};
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export const TextInput = styled.TextInput`
  width: 100%;
  margin-top: 4px;
  color: ${({ theme }) => theme.colors.background};
  font-size: 16px;
  font-family: ${({ theme }) => theme.fonts.regular};
`;

export const ButtonInput = styled(Button)`
  width: 85px;
`;
