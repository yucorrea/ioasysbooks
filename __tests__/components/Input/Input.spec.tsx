import React from 'react';
import { fireEvent } from "@testing-library/react-native";

import Input from '../../../src/components/Input';
import {
  matchSnapshotWithTheme,
  renderWithTheme,
} from '../../../src/helpers/functions/testing';

describe('Buttom Component', () => {

  it('matches snapshots', () => {
    const mockFunction = jest.fn();

    const snap = matchSnapshotWithTheme(
        <Input
          placeholder='Seu e-mail'
          label='E-mail'
          onChangeText={mockFunction}
          value='email@email.com'
          enableButton
          onPress={mockFunction}
        />
    );
    expect(snap).toMatchSnapshot();
  });

  it('render correctly', () => {
    const mockFunction = jest.fn();

    renderWithTheme(
      <Input
        placeholder='Seu e-mail'
        label='E-mail'
        onChangeText={mockFunction}
        value='email@email.com'
      />
    );

  });

  it('value change on input text', () => {
    const mockFunction = jest.fn();
    const { getByTestId } = renderWithTheme(
      <Input
        testID='input-email'
        placeholder='Seu e-mail'
        label='E-mail'
        onChangeText={mockFunction}
      />
    );

    const input = getByTestId('input-email');
    fireEvent.changeText(input, 'email@example.com');
    expect(mockFunction).toHaveBeenCalledWith('email@example.com');
  })

  it('check if contain button in input when has props', () => {
    const mockFunction = jest.fn();
    const { getByText } = renderWithTheme(
      <Input
        testID='input-password'
        placeholder='Senha'
        label='Senha'
        onChangeText={mockFunction}
        onPress={mockFunction}
        enableButton
      />
    );

    const button = getByText('Entrar')
    expect(button).toBeTruthy();
  })

});
