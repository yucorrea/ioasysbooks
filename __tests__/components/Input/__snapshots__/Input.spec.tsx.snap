// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttom Component matches snapshots 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.32)",
        "borderRadius": 4,
        "flexDirection": "row",
        "marginBottom": 16,
        "paddingBottom": 8,
        "paddingLeft": 13,
        "paddingRight": 16,
        "paddingTop": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.5)",
            "fontFamily": "Heebo-Regular",
            "fontSize": 12,
            "width": "100%",
          },
        ]
      }
    >
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Seu e-mail"
      placeholderTextColor="rgba(255,255,255,0.5)"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Heebo-Regular",
            "fontSize": 16,
            "marginTop": 4,
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="email@email.com"
    />
  </View>
  <View
    animatedStyle={
      Object {
        "value": Object {
          "height": 36,
          "width": 86,
        },
      }
    }
    collapsable={false}
    style={
      Object {
        "height": 36,
        "width": 86,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 44,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 1,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "color": "#B22E6F",
            "fontFamily": "Heebo-Medium",
            "fontSize": 16,
            "opacity": 1,
          }
        }
      >
        Entrar
      </Text>
    </View>
  </View>
</View>
`;

exports[`Buttom Component matches snapshots 2`] = `undefined`;
